### PR TITLE
Canonical override

### LINKED ISSUE #1314: Build canonical override

### PR DESCRIPTION
**Pull request by sanderson**
Closes #1314

Adds the ability to override the auto-generated canonical URL

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

**Comment by pierwill** [comment]
Is there a page that is best suited to testing this functionality?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -644,6 +645,25 @@ Redoc generates HTML documentation using the InfluxDB `swagger.yml`.
 For more information about generating InfluxDB API documentation, see the
 [API Documentation README](https://github.com/influxdata/docs-v2/tree/master/api-docs#readme).
 
+## Canonical URLs
+Search engines use canonocial URLs to accurately rank pages with similar or identical content.
```
Comment (noramullen1) [inline CONTRIBUTING.md:649]: extra syllable in canonical here

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -644,6 +645,25 @@ Redoc generates HTML documentation using the InfluxDB `swagger.yml`.
...
+The `canonical` HTML meta tag identifies which page should be used as the source of truth.
+
+By default, canonical URLs are automatically generated for each page in the InfluxData
+documentation using the latest version of the current product and the current path.
+
+Use the `canonical` frontmatter to override the auto-generated canonical URL.
+
+_**Note:** The `canonical` front matter supports the `{{< latest >}}` shortcode._
```
Comment (noramullen1) [inline CONTRIBUTING.md:657]: frontmatter

**Comment by sanderson** [comment]
@pierwill The only way to test it is to run the server, inspect the HTML `<head>` and look for the `<link rel="canonical" href="...">` meta tag.

**Review by noramullen1 — APPROVED** [review]
just a couple little typos

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -644,6 +645,25 @@ Redoc generates HTML documentation using the InfluxDB `swagger.yml`.
...
 [API Documentation README](https://github.com/influxdata/docs-v2/tree/master/api-docs#readme).
 
+## Canonical URLs
+Search engines use canonocial URLs to accurately rank pages with similar or identical content.
+The `canonical` HTML meta tag identifies which page should be used as the source of truth.
+
+By default, canonical URLs are automatically generated for each page in the InfluxData
+documentation using the latest version of the current product and the current path.
+
+Use the `canonical` frontmatter to override the auto-generated canonical URL.
```
Comment (kelseiv) [inline CONTRIBUTING.md:655]: Is using the latest shortcode the only reason to override the auto-generated cononical URL? If yes, perhaps:
To use the `{{< latest >}}` shortcode, add the `canonical` frontmatter to override the auto-generated canonical URL.

Do we expand on why folks would use the latest shortcode somewhere?

File: content/influxdb/v1.3/write_protocols/line_protocol_reference.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ menu:
   influxdb_1_3:
     weight: 10
     parent: Write protocols
+canonical: /{{< latest "influxdb" "v2" >}}/reference/syntax/line-protocol/
```
Comment (kelseiv) [inline content/influxdb/v1.3/write_protocols/line_protocol_reference.md:9]: Scott, did you mean to add this only to line protocol reference docs?

**Review by kelseiv — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (sanderson) [reply CONTRIBUTING.md:655]: @kelseiv No, there are two use cases for this functionality:

1. If the canonical content is on a page that is different than the auto-generated canonical URL. For example, let's say we create the following page:

    `/influxdb/v2.0/this/page/`

    When we cut a v2.1 release, the autogenerated canonical URL for this page will be:

    `/influxdb/v2.1/this/page`

    BUT, what if we change the location of the page in v2.1 to `/influxdb/v2.1/that/page/` ? The canonical URL on `/influxdb/v2 will point to a page that doesn't exist or gets redirected. We can override it to point to the new page.

2. We can begin to pass the SEO strength of the 1.x docs to the v2 docs by pointing canonical links to v2 docs.

File: content/influxdb/v1.3/write_protocols/line_protocol_reference.md (hunk 4 above)
Comment (sanderson) [reply content/influxdb/v1.3/write_protocols/line_protocol_reference.md:9]: Yes, just for now. We will need to identify other pages that will benefit from this. This was the page I thought of first and used to test. We certainly will want to add this frontmatter to other pages.

**Review by pierwill — APPROVED** [review]
👍

**Pull request merged**